virtual_machine.go

```diff
@@ -495,21 +497,23 @@ func (v *VirtualMachine) AgentExecStatus(pid int) (status *AgentExecStatus, err
 }
 
 func (v *VirtualMachine) WaitForAgentExecExit(pid, seconds int) (*AgentExecStatus, error) {
-	wait := time.Duration(seconds) * time.Second
+	timeout := time.After(time.Duration(seconds) * time.Second)
+	ticker := time.NewTicker(DefaultWaitInterval)
+	defer ticker.Stop()
+
 	for {
+		status, err := v.AgentExecStatus(pid)
+		if err != nil {
+			return nil, err
+		}
+		if status.Exited {
+			return status, nil
+		}
+
 		select {
-		case <-time.After(wait):
```

Comment:
the timeout would reset on each iteration of the loop, so would never trigger because of the default .. which would be very busy spin?

Reply:
so this is the only confusing change. for sure it needs a ticker or a sleep as that was over looked but why would the wait duration be reset every loop? i feel this just hammered the API checking the pid over and over but still should have timed out after duration. i love the ticker change and this should still go in i just want to know what im missing that the wait duration was being reset.

Reply:
see here https://go.dev/play/p/C9sjvZf-MKx
<img width="517" alt="image" src="https://user-images.githubusercontent.com/1652176/231848452-509a1bf0-bc08-4879-bb9e-e40ed06fdf8b.png">

Reply:
basically, the `time.After` inside a `select` is effectively evaluated on entry to the select.

Reply:
welp obviously i didnt test this